File: app/src/modules/evaluation.ts
class Evaluation {
	name: string
	weight: number
	score: [number, number]
	average: [number, number]

	constructor(cells: any[]) {
		this.name = ""
		this.weight = 0
		this.score = [0, 0]
		this.average = [0, 0]
	}

	getScore(score: string): [number, number] {
		const regex = /(\d+,\d+?)(?:\s\/\s)(\d+)/
		const r = regex.exec(score)
		return [parseFloat(r![1].replace(",", ".")), parseFloat(r![2])]
	}

	getGrade(): number {
		return (this.score[0] / this.score[1]) * 100
	}

	getAverage(): number {
		return (this.average[0] / this.average[1]) * 100
	}

	json(): Object {
		const data = {
			name: this.name,
			weight: this.weight,
			user: {
				score: `${this.score[0]}/${this.score[1]}`,
				grade: this.getGrade(),
			},
			class: {
				score: `${this.average[0]}/${this.average[1]}`,
				grade: this.getAverage(),
			},
		}
		return data
	}
}
export default Evaluation

File: api/evaluation.py
import re


class Evaluation:
    def __init__(self, cells):
        self.absent = False
        self.name = cells[2].text.strip()
        self.weight = int(cells[3].text)
        self.score = self.get_score(cells[7].text)
        self.average = self.get_score(cells[7].text)

    def get_score(self, score):
        r = re.match(r"(\d+(,\d+)?)(?:\s\/\s)(\d+)", score)
        if not r:
            self.absent = True
            return (score, score)
        return (float(re.sub(",", ".", r.group(1))), float(r.group(3)))

    def get_grade(self):
        if self.absent:
            return "AS"
        return (self.score[0] / self.score[1]) * 100

    def get_average(self):
        return (self.average[0] / self.average[1]) * 100

    def to_dict(self):
        data = {
            "name": self.name,
            "weight": self.weight,
            "user": {
                "score": f"{self.score[0]}/{self.score[1]}",
                "grade": self.get_grade(),
            },
            "class": {
                "score": f"{self.average[0]}/{self.average[1]}",
                # "grade": self.get_average(),
            },
        }
        return data

    def __repr__(self):
        return f"{self.name}, {self.weight}, {self.score}, {self.average}"
